app/src/controllers/ping.js

```diff
@@ -41,9 +45,16 @@ const addPing = async (req, res, next) => {
 
     console.log(runData)
     if (event === 'solo') {
-      // alter solo job queue
-      const res = await dbAddRun(runData);
-      console.log(res)
+      const delay = nextScheduledRun(monitor.schedule) + monitor.grace_period + monitor.tolerable_runtime;
```

Comment:
Have you checked that this works? `nextScheduleRun` returns a date object I think and `grace_period` and `tolerable_runtime` are measured in seconds.
For my code I used 
```
const calculateDelay = (monitor) => {
  const runTime = nextScheduledRun(monitor.schedule)._date.ts +
    ((monitor.grace_period + monitor.tolerable_runtime) * 1000); // milliseconds from epoch

  return (runTime - Date.now()) / 1000; // delay in seconds
};
```

Reply:
I realized after I committed that I forgot to test this, and then found out that my db schema wasn't updated because I hadn't deleted the volume from before. Sorry about that!

Reply:
No worries 😄